pkg/azure/client.go

```diff
@@ -96,8 +105,31 @@ func (az *azClient) GetPublicIP(resourceID string) (n.PublicIPAddress, error) {
 }
 
 // DeployGateway is a method that deploy the appgw and related resources
-func (az *azClient) DeployGateway(subnetID string) (err error) {
+func (az *azClient) DeployGatewayWithVnet(resourceGroupName ResourceGroup, vnetName ResourceName, subnetName ResourceName, subnetPrefix string) (err error) {
+	vnet, err := az.getVnet(resourceGroupName, vnetName)
+	if err != nil {
+		return
+	}
+
+	glog.Infof("Checking the Vnet %s for a subnet with prefix %s", vnetName, subnetPrefix)
+	subnet, err := az.findSubnet(vnet, subnetPrefix)
+	if err != nil {
+		glog.Infof("Unable to find a subnet. Creating a subnet %s with prefix %s in Vnet %s", subnetName, subnetPrefix, vnetName)
+		subnet, err = az.createSubnet(vnet, subnetName, subnetPrefix)
+		if err != nil {
+			return
```

Comment:
return `err` ?

Reply:
This is so mysterious. Return type of this function is `(err error)`